File: Chapter05/Java/StorageExample3/app/src/main/java/net/zenconsult/android/StoreData.java
package net.zenconsult.android;

import android.content.ContentValues;
import android.database.sqlite.SQLiteDatabase;
import android.util.Log;

import net.zenconsult.android.crypto.Crypto;

public class StoreData {

    private static final String TAG = "StoreData";

    public static long store(Crypto crypto, ContactsDb db, Contact contact) {
        // Prepare values
        ContentValues values = new ContentValues();
        try {
            values.put("FIRSTNAME", crypto.armorEncrypt(contact.getFirstName().getBytes()));
            values.put("LASTNAME", crypto.armorEncrypt(contact.getLastName().getBytes()));
            values.put("EMAIL", crypto.armorEncrypt(contact.getEmail().getBytes()));
            values.put("PHONE", crypto.armorEncrypt(contact.getPhone().getBytes()));
            values.put("ADDRESS1", crypto.armorEncrypt(contact.getAddress1().getBytes()));
            values.put("ADDRESS2ADDRESS2", crypto.armorEncrypt(contact.getAddress2().getBytes()));
        } catch (Exception e) {
            Log.e(TAG, "store=>error: ", e);
        }

        SQLiteDatabase wdb = db.getWritableDatabase();
        return wdb.insert(ContactsDb.TABLE_NAME, null, values);
    }
}

File: Chapter05/Kotlin/StorageExample3/app/src/main/java/net/zenconsult/android/StoreData.kt
package net.zenconsult.android

import android.content.ContentValues
import net.zenconsult.android.crypto.Crypto

class StoreData {

    companion object {

        fun store(crypto: Crypto, db: ContactsDb, contact: Contact): Long {
            val values = ContentValues()
            values.apply {
                put("FIRSTNAME", crypto.armorEncrypt(contact.firstName.toByteArray()))
                put("LASTNAME", crypto.armorEncrypt(contact.lastName.toByteArray()))
                put("EMAIL", crypto.armorEncrypt(contact.email.toByteArray()))
                put("PHONE", crypto.armorEncrypt(contact.phone.toByteArray()))
                put("ADDRESS1", crypto.armorEncrypt(contact.address1.toByteArray()))
                put("ADDRESS2", crypto.armorEncrypt(contact.address2.toByteArray()))
            }
            return db.writableDatabase.insert(ContactsDb.TABLE_NAME, null, values)
        }
    }
}